### PR TITLE
🌱 Uplift k8s to 1.24.1 and support new control plane taint

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
Uplift k8s to 1.24.1

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
Will be removed, this is to test CI behaviour. 

/hold

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]
This is missing to address this https://github.com/metal3-io/metal3-dev-env/issues/898#issuecomment-1018457872 comment

**Comment by adilGhaffarDev** [comment]
> This is missing to address this [metal3-io/metal3-dev-env#898 (comment)](https://github.com/metal3-io/metal3-dev-env/issues/898#issuecomment-1018457872) comment

I am adding this.

File: scripts/environment.sh [hunk 1]
```diff
@@ -50,7 +50,7 @@ fi
 
 export FROM_K8S_VERSION="v1.23.5"
```
Comment (furkatgofurov7) [inline scripts/environment.sh:51]:
```suggestion
export FROM_K8S_VERSION="v1.23.8"
```

File: test/e2e/config/e2e_conf.yaml [hunk 2]
```diff
@@ -109,7 +109,7 @@ providers:
 variables:
   CNI: "/tmp/calico.yaml"
   KUBERNETES_VERSION: "v1.23.5"
```
Comment (furkatgofurov7) [inline test/e2e/config/e2e_conf.yaml:111]:
```suggestion
  KUBERNETES_VERSION: "v1.23.8"
```

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Pull request merged**